distod/src/main/scala/com/github/codelionx/distod/actors/master/Master.scala

```diff
@@ -104,11 +113,24 @@ class Master(context: ActorContext[Command], stash: StashBuffer[Command], localP
         // TODO: remove
 //         testPartitionMgmt()
 
-        val state = rootCandidateState ++ L1candidateState
+        state ++= rootCandidateState ++ L1candidateState
+
+        // prepare L2 nodes:
+        val suffixCandidateSets = attributes.map(CandidateSet.from(_))
+        for(l1Node <- L1candidateState.keySet) {
+          val prefixedState = state.withPrefix(l1Node)
+          val prefixId = prefixedState(CandidateSet.empty).id
+          for(a <- attributes if a > prefixId.head) {
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Usage of head on collections.](https://app.codacy.com/app/CodeLionX/distod/pullRequest?prid=4697513)